File: app/src/main/java/com/anokmik/databinding/communicator/ObservableGreetingFragmentCommunicator.java
package com.anokmik.databinding.communicator;

import android.databinding.adapters.TextViewBindingAdapter;

import com.anokmik.databinding.model.ObservableGreeting;

public class ObservableGreetingFragmentCommunicator {

    public ObservableGreeting observableGreeting = new ObservableGreeting();

    public TextViewBindingAdapter.OnTextChanged onGreetingTextChanged = new TextViewBindingAdapter.OnTextChanged() {
        @Override
        public void onTextChanged(CharSequence s, int start, int before, int count) {
            observableGreeting.name.set(s.toString());
        }
    };

}

File: app/src/main/kotlin/com/anokmik/databinding/communicator/ObservableGreetingFragmentCommunicator.kt
package com.anokmik.databinding.communicator;

import android.databinding.adapters.TextViewBindingAdapter
import com.anokmik.databinding.model.ObservableGreeting;

class ObservableGreetingFragmentCommunicator {

    @JvmField
    val observableGreeting: ObservableGreeting = ObservableGreeting()

    @JvmField
    val onGreetingTextChanged: TextViewBindingAdapter.OnTextChanged = TextViewBindingAdapter.OnTextChanged { s, start, before, count -> observableGreeting.name.set(s.toString()) }

}